runtime/src/main/scala/akka/grpc/Grpc.scala

```diff
@@ -5,85 +5,135 @@
 package akka.grpc
 
 import akka.NotUsed
+import akka.grpc.GrpcProtocol.{ GrpcProtocolMarshaller, _ }
+import akka.http.javadsl.{ model => jmodel }
+import akka.http.scaladsl.model.HttpEntity.{ Chunk, ChunkStreamPart, LastChunk }
 import akka.http.scaladsl.model._
 import akka.stream.Attributes
 import akka.stream.impl.io.ByteStringParser
-import akka.stream.impl.io.ByteStringParser.ParseResult
-import akka.stream.impl.io.ByteStringParser.ParseStep
+import akka.stream.impl.io.ByteStringParser.{ ByteReader, ParseResult, ParseStep }
 import akka.stream.scaladsl.Flow
 import akka.stream.stage.GraphStageLogic
 import akka.util.ByteString
 import io.grpc.{ Status, StatusException }
 
-object Grpc {
-  val contentType = MediaType.applicationBinary("grpc+proto", MediaType.NotCompressible).toContentType
-
-  // Flag to signal the start of an uncompressed frame
-  val notCompressed = ByteString(0)
-  // Flag to signal the start of an frame compressed according to the Message-Encoding from the header
-  val compressed = ByteString(1)
+/**
+ * Implementation of the gRPC protocol.
+ */
+object Grpc extends GrpcVariant {
```

Comment:
It was initially unclear to me that this class is intended to contain logic that is specific to the native gRPC protocol, and not things that are generic for across native grpc, grpc-web and grpc-web-text. Perhaps have 'Native' in the name?

Reply:
I mostly agree - in this case grpc defines most of the protocol used by grpc-web variants.
Could possibly split the shared/common stuff out into a separate object and have the variants only have the specifics for their concrete protocols.

Reply:
> Could possibly split the shared/common stuff out into a separate object and have the variants only have the specifics for their concrete protocols

Yes, I think it would be helpful to have shared/common stuff in `Grpc` and the parts that are specific for the native protocol in something like `GrpcProtocolNative` (next to `GrpcProtocolWeb` etc).

Reply:
I've separated out the common parts as much as sensible (I hope) and moved the protocol definitions into .internal.